lib/FusionInventory/Agent/Task/Inventory/Linux/Networks.pm

```diff
@@ -128,31 +128,35 @@ sub _getInterfaces {
             );
         }
 
-        if (-r "/sys/class/net/$interface->{DESCRIPTION}/speed") {
-            my $speed = getFirstLine(
-                file => "/sys/class/net/$interface->{DESCRIPTION}/speed"
-            );
-            $interface->{SPEED} = $speed if $speed;
-        }
-        # On older kernels, we should try ethtool system call for speed
-        if (!$interface->{SPEED}) {
-            $logger->debug("looking for interface speed from syscall:");
-            my $infos = getInterfacesInfosFromIoctl(
-                interface => $interface->{DESCRIPTION},
-                logger    => $logger
-            );
-            if ($infos->{SPEED}) {
-                $logger->debug_result(
-                    action => 'retrieving interface speed from syscall',
-                    data   => $infos->{SPEED}
+        if (defined($interface->{STATUS}) && $interface->{STATUS} eq 'Up') {
+            if (-r "/sys/class/net/$interface->{DESCRIPTION}/speed") {
+                my $speed = getFirstLine(
+                    file => "/sys/class/net/$interface->{DESCRIPTION}/speed"
                 );
-                $interface->{SPEED} = $infos->{SPEED};
-            } else {
-                $logger->debug_result(
-                    action => 'retrieving interface speed from syscall',
-                    status => 'syscall failed'
+                $interface->{SPEED} = $speed if $speed;
+            }
+            # On older kernels, we should try ethtool system call for speed
+            if (!$interface->{SPEED}) {
+                $logger->debug("looking for interface speed from syscall:");
+                my $infos = getInterfacesInfosFromIoctl(
+                    interface => $interface->{DESCRIPTION},
+                    logger    => $logger
                 );
+                if ($infos->{SPEED}) {
+                    $logger->debug_result(
+                        action => 'retrieving interface speed from syscall',
+                        data   => $infos->{SPEED}
+                    );
+                    $interface->{SPEED} = $infos->{SPEED};
+                } else {
+                    $logger->debug_result(
+                        action => 'retrieving interface speed from syscall',
+                        status => 'syscall failed'
+                    );
+                }
             }
+        } else {
+            $interface->{SPEED} = 0;
```

Comment:
Finally why setting SPEED in the case STATUS is not 'Up' ?

Reply:
I made it to update/reset incorrect speed of the downed interfaces in the database.

Reply:
Okay I understand. You should add a comment then so we know why we do that.

Reply:
Done.